packages/react-portal-compat/src/PortalCompatProvider.test.tsx

```diff
@@ -0,0 +1,68 @@
+import { ThemeClassNameContext } from '@fluentui/react-shared-contexts';
+import { FluentProvider } from '@fluentui/react-provider';
+import { resetIdsForTests } from '@fluentui/react-utilities';
+import { renderHook } from '@testing-library/react-hooks';
+import * as React from 'react';
+
+import { usePortalCompat } from './PortalCompatContext';
+import { PortalCompatProvider } from './PortalCompatProvider';
+
+describe('PortalCompatProvider', () => {
+  afterEach(() => {
+    resetIdsForTests();
+  });
+
+  it('registers a function in a context', () => {
+    const { result } = renderHook(() => usePortalCompat(), { wrapper: PortalCompatProvider });
+
+    expect(result.current).toBeInstanceOf(Function);
+  });
+
+  it('during register adds a className from "ThemeClassNameContext" context', () => {
+    const element = document.createElement('div');
+    const { result } = renderHook(() => usePortalCompat(), {
+      wrapper: props => (
+        <FluentProvider theme={{ colorNeutralBackground1: '#ccc' }}>
+          <PortalCompatProvider>{props.children}</PortalCompatProvider>
+        </FluentProvider>
+      ),
+    });
+
+    expect(result.current(element)).toBeInstanceOf(Function);
+    expect(element.classList.length).toBe(1);
+    expect(element.classList.item(0)).toMatch(/fui-FluentProvider\d/);
+  });
+
+  it('during unregister remove a className from "ThemeClassNameContext" context', () => {
+    const element = document.createElement('div');
+
+    const { result } = renderHook(() => usePortalCompat(), {
+      wrapper: props => (
+        <FluentProvider theme={{ colorNeutralBackground1: '#ccc' }}>
+          <PortalCompatProvider>{props.children}</PortalCompatProvider>
+        </FluentProvider>
+      ),
+    });
+    const unregister = result.current(element);
+
+    expect(element.classList.length).toBe(1);
+    expect(unregister).toBeInstanceOf(Function);
+
+    expect(unregister()).toBeUndefined();
+    expect(element.classList.length).toBe(0);
```

Comment:
**suggestion:** can we maybe use inlineSnapshot or explicit check for particular class names ?

Reply:
I left `expect(element.classList.length).toBe(0);` as it makes sense (at least for me), but in other places I used snapshots for assertions (see `dd7df4e`)